public/app/features/templating/variable_srv.ts

```diff
@@ -130,6 +137,7 @@ export class VariableSrv {
     return this.$q.all(promises).then(() => {
       if (emitChangeEvents) {
         this.$rootScope.$emit('template-variable-value-updated');
+        this.ignoreRefresh = true;
         this.$rootScope.$broadcast('refresh');
```

Comment:
a better way to solve this without using the instance variable/state ignoreRefresh would be to send an object in the refresh event like:
this.$rootScope.$broadcast('refresh', {fromVariableValueUpdated: true});

Then you can in the onDashboardRefresh function take this argument in (evt, payload)

Reply:
Thanks, feedback!